### PR TITLE
Update

### PR DESCRIPTION
**Pull request by helloWorldCindy**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #none# - documentation fixes and/or test additions
- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

Upgrade Ember CLI version from 2.11.0 to 2.12.3

### CODE REVIEW

**Comment by juwara0** [comment]
DO NOT MERGE - This is still being worked on

File: .npmignore [hunk 1]
```diff
@@ -9,6 +9,7 @@
 .ember-cli
 .eslintignore
 .eslintrc
```
Comment (juwara0) [inline .npmignore]: We can remove the `.eslintrc` entry as well as the `.eslintrc` file.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module.exports = {
```
Comment (notmessenger) [inline .eslintrc.js:1]: This files content should only be:

```
/* eslint-env node */

module.exports = {
  extends: 'frost-standard',
}
```

File: .npmignore [hunk 3]
```diff
@@ -8,10 +8,11 @@
 .editorconfig
 .ember-cli
 .eslintignore
-.eslintrc
+.eslintrc.js
 .github
 .gitignore
 .jshintrc
+.eslintrc.js
```
Comment (notmessenger) [inline .npmignore]: Remove this duplicate entry

File: .travis.yml [hunk 4]
```diff
@@ -33,6 +33,7 @@ matrix:
   fast_finish: true
   allow_failures:
   - env: EMBER_TRY_SCENARIO=ember-release
+  - node_js: 'stable'
```
Comment (notmessenger) [inline .travis.yml]: This entry, and section, does not match https://github.com/ciena-frost/ember-frost-test/blob/master/.travis.yml, for example.  @juwara0 how/when are we handling the standardization of these configurations?  At the very least I'd rather not be introducing this change even if we're not going to remove the entire section for standardization (if that is the action we decided to take)

File: ember-cli-build.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-/* global require, module */
+/* eslint-env node */
 var EmberAddon = require('ember-cli/lib/broccoli/ember-addon')
```
Comment (notmessenger) [inline ember-cli-build.js]: This line should be changed to

`const EmberAddon = require('ember-cli/lib/broccoli/ember-addon')`

File: .npmignore [hunk 6]
```diff
@@ -8,10 +8,11 @@
 .editorconfig
 .ember-cli
 .eslintignore
-.eslintrc
+.eslintrc.js
 .github
 .gitignore
 .jshintrc
```
Comment (notmessenger) [inline .npmignore]: Remove this entry

File: .eslintrc [hunk 7]
```diff
@@ -1,6 +0,0 @@
-{
-  "extends": ["frost-standard"],
-  "globals": {
-    "URL": false
```
Comment (notmessenger) [inline .eslintrc:4]: Determine whether the global `URL` value needs to be added to the _.eslintrc.js_ file

**Review by notmessenger — CHANGES_REQUESTED** [review]

**Comment by notmessenger** [comment]
- The contents of the _tests/helpers/start-app.js_ file need to be changed to:

```
import Ember from 'ember'
const {merge, run} = Ember

import Application from '../../app'
import config from '../../config/environment'

export default function startApp (attrs) {
  let attributes = merge({}, config.APP)
  attributes = merge(attributes, attrs) // use defaults, but you can override;

  return run(() => {
    let application = Application.create(attributes)
    application.setupForTesting()
    application.injectTestHelpers()
    return application
  })
}
```

- The _tests/.eslintrc_ file needs to be deleted

- the `ember-cli-app-version` dependency should be removed from _package.json_

- the `ember-cli-test-loader` dependency should be removed from _package.json_

- should set the `broccoli-asset-rev` dependency in _package.json_ to `^2.4.5` so that we're in alignment with the value from ember-cli@2.12.3 and the other repos being set to the same

- should set the `ember-cli-inject-live-reload` dependency in _package.json_ to `^1.4.1` so that we're in alignment with the value from ember-cli@2.12.3 and the other repos being set to the same

- should set the `ember-export-application-global` dependency in _package.json_ to `^1.0.5` so that we're in alignment with the value from ember-cli@2.12.3 and the other repos being set to the same

- should set the `ember-load-initializers` dependency in _package.json_ to `^0.6.0` so that we're in alignment with the value from ember-cli@2.12.3 and the other repos being set to the same

File: .travis.yml (hunk 4 above)
Comment (helloWorldCindy) [reply .travis.yml]: Should I changed it to be the same as the one in  https://github.com/ciena-frost/ember-frost-test/blob/master/.travis.yml ? But the builds are failing with Node.js: stable

File: .travis.yml [hunk 8]
```diff
@@ -22,18 +22,17 @@ cache:
...
     - CXX=g++-4.8
     - secure: oeu9jlEjQaqNMt1G7i73ciojX9+lCNoQOFjN20hqIK5uUkBOmDlwvo5hvcOzDj4IPpScgMaXYbKGolZSNCNZ3XKecMLyCSWuD1wm51HfeIU/nblMtlwJQUVM7bJBxua8m9NyCbXLbqlkfG8cNvvReJECZ9aheMYB/FD3zxAzCH9xvP3q3xGE4QRCtTM+Pz3WXpyUtQFxQqXxuz4wPFybZL2aTeMfti3HIz4VzOooJjp9k/MLkGR29qg7ivkfAGY3PB0U8b5VaojOAkMLwgcpMbfgrt9V66aqE0dOfh+L2/2Vmii34ZKh+R1+lFuWeait+cyzdqgMn6nTE++Ltj36+Yfk+R5xj2pl9/rQ2tLeaXzNznF9XM+V7tAf+DaYiZs9RSJNA+Pi/aL2MRv3pMu6G1bgJSeIBpNTZQxOSPbK1gOwiCexuTBgLAc+I9+pyflkEpbWNubXbTCuP3jfL1GN9gBwMAo6V7aevBSlgUfvt+g3pL7NIwynoR3QE+PziiE9sPk2sEDjoV0r5zDx/0sztrgRSiA0Tj3PiTeOl7qFbH4Sl4hvKWWkkCcLLUumQyvOUh+x58LnYddtUDwn5MeY7cA1/+XqhjwBmeJjfaFJMniIiS1ZYdhh6T9yiEH02m5801JQtQn6VdKwl4t1OQa8ZDXOmnenSVPz7lItkMCgFw4=
 
 matrix:
   fast_finish: true
-  allow_failures:
-  - env: EMBER_TRY_SCENARIO=ember-release
-
+  gallow_failures:
+     - env: EMBER_TRY_SCENARIO=ember-release
```
Comment (dafortin) [inline .travis.yml]: We should remove this. It's not necessary. 

You will need to allow failures on:
```
- node_js: 'stable'
```

Comment (dafortin) [reply .travis.yml]: `gallow_failures`. I thought that It should be `allow_failures`.

File: package.json [hunk 9]
```diff
@@ -26,27 +26,25 @@
...
     "ember-graphlib": "^2.2.0",
-    "ember-load-initializers": "^0.6.3",
+    "ember-load-initializers": "^0.6.0",
     "ember-lodash-shim": "^2.0.0",
     "ember-resolver": "^2.1.1",
-    "ember-source": "~2.11.2",
+    "ember-source": "~2.12.0",
     "ember-test-utils": "^1.11.0",
-    "loader.js": "^4.1.0"
+    "loader.js": "^4.2.3"
```
Comment (dafortin) [inline package.json]: `ember-cli-babel` should be `^5.1.7`

File: .travis.yml (hunk 8 above)
Comment (helloWorldCindy) [reply .travis.yml]: Fixed.

File: package.json (hunk 9 above)
Comment (helloWorldCindy) [reply package.json]: Fixed.

**Comment by notmessenger** [comment]
The `http` in the _.gitignore_ file should be `https`

File: .travis.yml [hunk 10]
```diff
@@ -22,18 +22,18 @@ cache:
...
     - CXX=g++-4.8
     - secure: oeu9jlEjQaqNMt1G7i73ciojX9+lCNoQOFjN20hqIK5uUkBOmDlwvo5hvcOzDj4IPpScgMaXYbKGolZSNCNZ3XKecMLyCSWuD1wm51HfeIU/nblMtlwJQUVM7bJBxua8m9NyCbXLbqlkfG8cNvvReJECZ9aheMYB/FD3zxAzCH9xvP3q3xGE4QRCtTM+Pz3WXpyUtQFxQqXxuz4wPFybZL2aTeMfti3HIz4VzOooJjp9k/MLkGR29qg7ivkfAGY3PB0U8b5VaojOAkMLwgcpMbfgrt9V66aqE0dOfh+L2/2Vmii34ZKh+R1+lFuWeait+cyzdqgMn6nTE++Ltj36+Yfk+R5xj2pl9/rQ2tLeaXzNznF9XM+V7tAf+DaYiZs9RSJNA+Pi/aL2MRv3pMu6G1bgJSeIBpNTZQxOSPbK1gOwiCexuTBgLAc+I9+pyflkEpbWNubXbTCuP3jfL1GN9gBwMAo6V7aevBSlgUfvt+g3pL7NIwynoR3QE+PziiE9sPk2sEDjoV0r5zDx/0sztrgRSiA0Tj3PiTeOl7qFbH4Sl4hvKWWkkCcLLUumQyvOUh+x58LnYddtUDwn5MeY7cA1/+XqhjwBmeJjfaFJMniIiS1ZYdhh6T9yiEH02m5801JQtQn6VdKwl4t1OQa8ZDXOmnenSVPz7lItkMCgFw4=
 
 matrix:
   fast_finish: true
   allow_failures:
-  - env: EMBER_TRY_SCENARIO=ember-release
-
+    - node_js: 'stable'
+
```
Comment (notmessenger) [inline .travis.yml]: Are we adding empty spaces here or  just an empty row?  The diff makes it look like spaces.

File: tests/.eslintrc.js [hunk 11]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+  env: {
+    embertest: true
+  }
+};
```
Comment (notmessenger) [inline tests/.eslintrc.js]: Remove `;`

File: tests/dummy/config/environment.js [hunk 12]
```diff
@@ -1,3 +1,5 @@
+/* eslint-env node */
+
 module.exports = function (environment) {
```
Comment (notmessenger) [inline tests/dummy/config/environment.js:3]: Set `locationType` to `hash`

**Review by notmessenger — CHANGES_REQUESTED** [review]

File: tests/dummy/config/environment.js (hunk 12 above)
Comment (juwara0) [reply tests/dummy/config/environment.js:3]: Fixed

**Review by notmessenger — APPROVED** [review]

**Comment by notmessenger** [comment]
👍

**Pull request merged**